src/index.ts

```diff
@@ -323,21 +312,29 @@ export default function solidPlugin(options: Partial<Options> = {}): Plugin {
         ? ['solid-js', 'solid-js/web', 'solid-js/store', 'solid-js/html', 'solid-js/h']
         : [];
 
-      const test =
-        userConfig.mode === 'test'
-          ? {
-              test: {
-                globals: true,
-                ...(options.ssr ? {} : { environment: 'jsdom' }),
-                transformMode: {
-                  [options.ssr ? 'ssr' : 'web']: [/\.[jt]sx?$/],
-                },
-                ...(isJestDomInstalled()
-                  ? { setupFiles: [require.resolve('@testing-library/jest-dom/extend-expect.js')] }
-                  : {}),
-              },
-            }
-          : {};
+      const test = (userConfig as any).test || {};
+
+      if (userConfig.mode === 'test') {
+        if (!test.environment && !options.ssr) {
+          test.environment = 'jsdom';
+        }
+        if (!test.transformMode) {
+          test.transformMode = { [options.ssr ? 'ssr' : 'web']: [/\.[jt]sx?$/] };
+        }
```

Comment:
The latest Vite removed `transformMode` for a new `testTransformMode`. The second one would use an array of strings instead of regex `['\.[jt]sx?$']`

Reply:
Ah, thanks, I didn't really keep up with the vitest development in the last months. I should fix that.

Reply:
It makes more sense to remove this, since it only overwrites what hasn't got environment defined in any case.